### PR TITLE
MMI-3071 Minor updates

### PR DESCRIPTION
**Pull request by laidaoyu**
- removed redundant locks
- used concurrent dictionary to store the local etag values
- updated async methods

### CODE REVIEW

**Review by Fosol — APPROVED** [review]

File: services/net/content/ContentManager.cs [hunk 1]
```diff
@@ -379,22 +373,27 @@ private void UpdateEtagLocalCache(string keyName, string? etag)
...
+            });
     }
 
     /// <summary>
     /// Get ingest service by topic.
     /// </summary>
     /// <param name="topic"></param>
     /// <returns></returns>
-    private API.Areas.Services.Models.Ingest.IngestModel? GetIngestsByTopic(string topic)
+    private async Task<API.Areas.Services.Models.Ingest.IngestModel?> GetIngestsByTopic(string topic)
```
[the remark this answers is not in the document]
Comment (laidaoyu) [reply services/net/content/ContentManager.cs]: updated. thanks

File: services/net/content/ContentManager.cs [hunk 2]
```diff
@@ -418,13 +414,10 @@ private void UpdateEtagLocalCache(string keyName, string? etag)
     /// </summary>
     /// <param name="code"></param>
     /// <returns></returns>
-    private API.Areas.Services.Models.Ingest.SourceModel? GetSource(string code)
+    private async Task<API.Areas.Services.Models.Ingest.SourceModel?> GetSource(string code)
```
[the remark this answers is not in the document]
Comment (laidaoyu) [reply services/net/content/ContentManager.cs]: updated. thanks

File: services/net/content/ContentManager.cs [hunk 3]
```diff
@@ -498,12 +491,9 @@ private void UpdateEtagLocalCache(string keyName, string? etag)
     /// If the request failed, get the data from local memory cache.
     /// </summary>
     /// <returns></returns>
-    private API.Areas.Editor.Models.Lookup.LookupModel? GetLookups()
+    private async Task<API.Areas.Editor.Models.Lookup.LookupModel?> GetLookups()
```
[the remark this answers is not in the document]
Comment (laidaoyu) [reply services/net/content/ContentManager.cs]: updated. thanks

File: services/net/content/ContentManager.cs [hunk 4]
```diff
@@ -433,7 +426,7 @@ private void UpdateEtagLocalCache(string keyName, string? etag)
     /// <typeparam name="T"></typeparam>
     /// <param name="keyName"></param>
     /// <returns></returns>
-    private T? GetLocalCacheList<T>(string keyName)
+    private async Task<T?> GetLocalCacheList<T>(string keyName)
```
[the remark this answers is not in the document]
Comment (laidaoyu) [reply services/net/content/ContentManager.cs]: updated. thanks

File: services/net/content/ContentManager.cs [hunk 5]
```diff
@@ -403,12 +402,9 @@ private void UpdateEtagLocalCache(string keyName, string? etag)
     /// If the request failed, get the data from local memory cache.
     /// </summary>
     /// <returns></returns>
-    private IEnumerable<API.Areas.Services.Models.Ingest.IngestModel>? GetIngests()
+    private Task<IEnumerable<API.Areas.Services.Models.Ingest.IngestModel>?> GetIngests()
```
[the remark this answers is not in the document]
Comment (laidaoyu) [reply services/net/content/ContentManager.cs]: updated. thanks

**Comment by Fosol** [comment]
I have updated the github action with a new version that I hope can handle .NET 9.0.  It appears that we can't test it until the next PR for some reason however.  I'm not certain why it's so difficult to test github actions.

**Pull request merged**